cmd/jujud/agent/machine.go

```diff
@@ -654,26 +655,17 @@ func (a *MachineAgent) ChangeConfig(mutate agent.ConfigMutator) error {
 // Backups.Restore: this will ensure that we can do all the file movements
 // required for restore and no one will do changes while we do that.
 // it will return error if the machine is already in this state.
-func (a *MachineAgent) PrepareRestore() error {
-	if a.restoreMode {
-		return errors.Errorf("already in restore mode")
-	}
+func (a *MachineAgent) PrepareRestore() {
 	a.restoreMode = true
-	return nil
 }
 
 // BeginRestore will flag the agent to disallow all commands since
 // restore should be running and therefore making changes that
 // would override anything done.
-func (a *MachineAgent) BeginRestore() error {
-	switch {
-	case !a.restoreMode:
-		return errors.Errorf("not in restore mode, cannot begin restoration")
-	case a.restoring:
-		return errors.Errorf("already restoring")
+func (a *MachineAgent) BeginRestore() {
```

Comment:
why are the checks gone?

Reply:
the error was never checked, and I think the errors that were there were too much (e.g. if the agent went down half way through, came back, then the state would be mucked up)

ultimately I want to get rid of all this, and move it into the workers